src/axom/quest/detail/clipping/MeshClipperImpl.hpp

```diff
@@ -818,39 +757,266 @@ class MeshClipperImpl : public MeshClipper::Impl
       axom::for_all<ExecSpace>(
         candidates.size(),
         AXOM_LAMBDA(axom::IndexType iCand) {
-          auto pieceId = candidatesView[iCand];
-          const axom::primal::Octahedron<double, 3>& geomPiece = geomOctsView[pieceId];
-
           auto tetIdId = candToTetIdIdView[iCand];
           auto tetId = tetIndices[tetIdId];
-          const auto& tet = meshTets[tetId];
 
-          const auto poly = primal::clip<double>(tet, geomPiece, EPS, tryFixOrientation);
+          auto cellId = tetId / NUM_TETS_PER_HEX;
+          auto pieceId = candidatesView[iCand];
+          const auto& meshTet = meshTets[tetId];
+          const OctahedronType& geomPiece = geomOctsView[pieceId];
 
-          if(poly.numVertices() >= 4)
+          double volume = 0.0;
+          LabelType tmpLabel =
+            computeMeshTetGeomPieceOverlap(meshTet, geomPiece, volume, screenLevel);
+          if(tmpLabel == LabelType::LABEL_IN || tmpLabel == LabelType::LABEL_ON)
           {
-            // Poly is valid
-            double volume = poly.volume();
-            SLIC_ASSERT(volume >= 0);
-            auto cellId = tetId / NUM_TETS_PER_HEX;
             RAJA::atomicAdd<ATOMIC_POL>(ovlap.data() + cellId, volume);
+            RAJA::atomicAdd<ATOMIC_POL>(contribCountPtr, std::int64_t(volume >= EPS));
+          }
+          if(tmpLabel == LabelType::LABEL_ON)
+          {
+            RAJA::atomicAdd<ATOMIC_POL>(clipCountPtr, std::int64_t(1));
           }
         });
     }
+    AXOM_ANNOTATE_END("MeshClipper:clipLoop_tetScreened");
+
+    axom::copy(&contribCount, contribCountPtr, sizeof(contribCount));
+    axom::copy(&clipCount, clipCountPtr, sizeof(clipCount));
+    axom::deallocate(contribCountPtr);
+    axom::deallocate(clipCountPtr);
 
+    SLIC_DEBUG(axom::fmt::format(""));
   }  // end of computeClipVolumes3DTets() function
 
+  /*!
+   * @brief Get the geometry in discrete pieces,
+   *   which can be tets or octs, and place them in a search tree.
+   * @return whether geometry are tetrahedra instead of octahedra.
```

Comment:
For clarity, please rephrase to something like
```suggestion
   * @return true if discrete geometry is composed of tetrahedra, false if octahedra.
```